router/core/http_transport_cancellation_test.go

```
@@ -0,0 +1,53 @@
+	t.Cleanup(server.Close)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, server.URL, nil)
+	require.NoError(t, err)
+
+	transport := newHTTPTransport(DefaultTransportRequestOptions(), nil, nil, "products", nil)
+	t.Cleanup(transport.CloseIdleConnections)
+
+	done := make(chan error, 1)
+	go func() {
+		_, roundTripErr := transport.RoundTrip(req)
```

golangci-lint (2.12.2)

[error] response body must be closed

(bodyclose)